mod.go

```diff
@@ -10,13 +10,8 @@ import (
 
 // Some utility functions
 
-func leadingZeros(x *Int) (z int) {
-	var t int
-	z = bits.LeadingZeros64(x[3])
-	t = bits.LeadingZeros64(x[2]); if z ==  64 { z = t +  64 }
-	t = bits.LeadingZeros64(x[1]); if z == 128 { z = t + 128 }
-	t = bits.LeadingZeros64(x[0]); if z == 192 { z = t + 192 }
-	return z
+func leadingZeros(x *Int) int {
```

Comment:
This method is not actually part of the 'functional' code, that is, code that is actually being used. It seems to be only used from `mod_test.go`, the method `TestLeadingZeroes`. 

So it's not worth optimizing, but since your implementation is a lot nicer, I think we should: 
 
- Not bother adding the new benchmark-tests (`BenchmarkBitLen`)
- Move `leadingZeroes` function into `mod_test.go`, to make it obvious that it's "supporting code" and not prod
- Make it use your new code

SGTY?

Reply:
I think we either remove `leadingZeroes` or make it public, since it's only used for `TestLeadingZeros`  in mod_test.go.
And BenchmarkBitLen is for benchmarking BitLen, not for leadingZeros.

Reply:
If we choose removing `leadingZeroes`, should I delete mod_test.go directly?

Reply:
Oh right, yes, let's just nuke it

Reply:
> And BenchmarkBitLen is for benchmarking BitLen, not for leadingZeros.

Ah, my bad. Then let's keep that, but I think we should perhaps, for each `N`, do a loop on all `0-256` bitsizes, instead of using just the four "magic" ones. Either that, or use random values.

Reply:
I think the 4 values can represent the 4 class of numbers well. And remember this is benchmarking, not testing.

Reply:
Yes, but if we add a skewed benchmark, someone might submit a "faster" implementation which is faster in those special cases, but not in the general case. In general it's good if we make the benchmarks somewhat generic.

Reply:
Since we remove `leadingZeroes`, I am ok with not adding `BenchmarkBitLen`.

Reply:
If we just use all bitsizes instead of only four (and clump it all into a single benchmark), that should be more or less a "perfect" sampling